cv_next/app/cv/[cvId]/components/commentSection/comment.tsx

```diff
@@ -28,69 +29,71 @@ export default function Comment({
   commentsOfComment = [],
   setCommentsOfComments,
 }: {
-  comment: CommentModel
-  userId: string
-  childDepth?: number
-  cv: CvModel
-  commentsOfComment: any[]
+  comment: CommentModel;
+  userId: string;
+  childDepth?: number;
+  cv: CvModel;
+  commentsOfComment: any[];
   setCommentsOfComments: (
     update: (prev: Map<string, any[]>) => Map<string, any[]>
-  ) => void
+  ) => void;
 }) {
   const [commentOnCommentStatus, setCommentOnCommentStatus] =
-    useState<boolean>(false)
-  const [commentOnComment, setCommentOnComment] = useState<string>("")
-  const [submitCommentOnComment, setSubmitCommentOnComment] =
-    useState<boolean>(false)
-
-  const [showAlert, setShowAlert] = useState<boolean>(false)
+    useState<boolean>(false);
+  const [commentOnComment, setCommentOnComment] = useState<string>("");
+  const router = useRouter();
+  const [showAlert, setShowAlert] = useState<boolean>(false);
   const onAlertClick = async (type: boolean) => {
-    if (type) await deleteCommentAction()
-    setShowAlert(false)
-  }
+    if (type) await deleteCommentAction();
+    setShowAlert(false);
+  };
+  const supabase = useSupabase();
 
   const addNewCommentClickEvent = async () => {
-    if (!userId) throw new Error("User not found")
-    const commentToAdd: NewCommentModel = {
-      data: commentOnComment,
-      document_id: cv.id,
-      parent_comment_Id: comment.id,
-      user_id: userId,
-    }
+    const userId = (await supabase.auth.getUser()).data.user?.id;
```

Comment:
We want to go through auth? Don't we want to call `getUser` api in hopes it's cached and doesn't need to go through supabase?

Reply:
Like the actual route itself? In theory we could, another option is to use getSession to get the data from the cookies

Reply:
Apparently we already get the user's id as a prop so this call is redundant... removed it completly